src/openms/source/FORMAT/PepXMLFile.cpp

```diff
@@ -1202,14 +1202,24 @@ namespace OpenMS
 
         optionalAttributeAsString_(aa_mod.aminoacid, attributes, "aminoacid");
         aa_mod.terminus = String(attributeAsString_(attributes, "terminus")).toLower();
-        if (aa_mod.terminus == "n")
+        // "y" if protein terminus, "n" if peptide terminus
+        aa_mod.protein_terminus = String(attributeAsString_(attributes, "protein_terminus")).toLower() == "y";
+        if (aa_mod.terminus == "n" && !aa_mod.protein_terminus)
         {
           term_spec = ResidueModification::N_TERM;
         }
-        else if (aa_mod.terminus == "c")
+        else if (aa_mod.terminus == "n" && aa_mod.protein_terminus)
```

Comment:
`aa_mod.terminus == "n"` is duplicated. Can you make the structure a bit clearer by using an internal if/else. same for C-terminus.